site/content/guides/claude-code-monitoring.md

```diff
@@ -0,0 +1,415 @@
+
+```bash
+2 failed, 5 passed, 7 total
+```
+*If you've followed every step in this tutorial, you may have different numbers for the number of tests run here.*
+
+
+When running a number of generated tests, the only issue we ran into was a failure with the message: `Error: expect.toBeVisible: Error: strict mode violation: getByText('Crime & Thrillers') resolved to 2 elements:`. As the error message implies, the issue is that this locator:
+
+```ts
+  await expect(page.getByText('Fantasy')).toBeVisible();
+```
+
+Found multiple results on the page. There are a number of ways to fix this with Claude Code. We could prompt Claude Code with:
+
+```{title="Prompt:"}
+revise locators in item-visibility.spec.ts to look at only the first result
+```
+
+This would change every locator, which may or may not be what you want! You could also paste in the error message to Claude and ask it to fix the script. Whatever route you take the result will be an edit to:
+
+```ts
+  await expect(page.getByText('Fantasy').first()).toBeVisible();
+```
+If you're having trouble reading the generated tests, take a look at [Checkly's Playwright documentation](https://www.checklyhq.com/learn/playwright/). Once Claude Code added a single `.first()` to the checks written above, everything was passing. This issue proved frequent enough that we added the following to Claude Code's permanent memory:
+
+```{title="Prompt:"}
+# When writing a Playwright check, if you find multiple elements that match a locator, make sure the locator includes .first() to take the first result.
+```
+This solved the general issue for new browser checks created by Claude Code.
+
+### 4. Going Further: Edit Multiple Checks and Configurations
+
+Another use case that may be useful for larger projects, try prompting Claude Code with something like:
+
+```{title="Prompt:"}
+update every `.check.ts` file for a check that touches [danube-web.shop](http://danube-web.shop/), and change the frequency to every 30 minutes.
+```
+
+Not only did this request work, but even produced sensible feedback about the checks that currently had no frequency setting:
+
+```{title="Reply from Claude Code:"}
+  The url.check.ts and api.check.ts files don't have individual frequency settings -
+  they inherit the default frequency from the project configuration in
+  checkly.config.ts.
+```
+
+Once we're done creating new checks, it's time to run them all.
+
+
+## Deploy With the Checkly CLI
+
+In our example, we want to deploy a couple of checks at once, so it's worth checking whether the configuration is properly formatted before deployment. Many problems would have been revealed while running `test`, but if, for example, the [logical IDs](https://www.checklyhq.com/docs/cli/constructs-reference/#synthetic-checks) of current and existing checks are colliding, we'll need to run `deploy` to detect that. Run the `deploy` command with the `-p` preview flag:
```

Comment:
The statement that only `deploy` detects colliding logical ids is incorrect.

<img width="703" height="138" alt="image" src="https://github.com/user-attachments/assets/509a75c1-dba5-406e-88f8-f33f8c623659" />

Reply:
whoa.gif

Didn't realize it would detect if there was collision not in your current tests but also with previously deployed ones. Edited to clarify we're really checking for project consistency.